libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,257 @@
   }
 
   @Test
   public void fasterRouteEngine_doesNotGetInitializedWithFalseOption() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder()
       .enableFasterRouteDetection(false)
       .build();
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, options, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
-    assertNull(navigation.getFasterRouteEngine());
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
+    assertNull(navigationWithOptions.getFasterRouteEngine());
   }
 
   @Test
   public void addMilestone_milestoneDidGetAdded() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
     Milestone milestone = new StepMilestone.Builder().build();
+
     navigation.addMilestone(milestone);
+
     assertTrue(navigation.getMilestones().contains(milestone));
   }
 
   @Test
   public void addMilestone_milestoneOnlyGetsAddedOnce() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
     navigationWithOptions.addMilestone(milestone);
     navigationWithOptions.addMilestone(milestone);
+
     assertEquals(1, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_milestoneDidGetRemoved() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
     navigationWithOptions.addMilestone(milestone);
-    assertEquals(1, navigationWithOptions.getMilestones().size());
     navigationWithOptions.removeMilestone(milestone);
+
     assertEquals(0, navigationWithOptions.getMilestones().size());
   }
 
   @Test
   public void removeMilestone_milestoneDoesNotExist() throws Exception {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().defaultMilestonesEnabled(false).build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     Milestone milestone = new StepMilestone.Builder().build();
```

Comment:
```
Milestone milestone = new StepMilestone.Builder().build();
navigationWithOptions.addMilestone(new StepMilestone.Builder().build());
```
Should be in the `Arrange` part.
We could take advantage of the naming here to understand clearer what's going on. Like naming `new StepMilestone.Builder().build()` as `inexistentMilestone` for example.